edb/common/uuidgen.py

```diff
@@ -18,12 +18,42 @@
 
 
 from __future__ import annotations
+from typing import *  # NoQA
 
+import hashlib
 import os
 import uuid
 
+from asyncpg.pgproto import pgproto
+
+
+UUID: Type[uuid.UUID] = pgproto.UUID
+
 
 def uuid1mc() -> uuid.UUID:
     """Generate a v1 UUID using a pseudo-random multicast node address."""
+
+    # Note: cannot use pgproto.UUID since it's UUID v1
     node = int.from_bytes(os.urandom(6), byteorder='little') | (1 << 40)
     return uuid.uuid1(node=node)
+
+
+# type-ignores below because the first argument to uuid.UUID is a string
+# called `hex` which is not something that pgproto.UUID supports.
+
+
+def uuid3(namespace: uuid.UUID, name: str) -> uuid.UUID:
```

Comment:
I'm just curious, could you use `-> UUID` instead of `uuid.UUID` here?

Reply:
"UUID" is seen as a type alias to uuid.UUID anyway. I wanted to make this explicit that Mypy does not see our UUIDs at all.

Reply:
Got it, thanks!